lib/ansible/modules/storage/emc/emc_vnx_mv_promote.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018, Luca 'remix_tj' Lorenzetto <lorenzetto.luca@gmail.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+from __future__ import (absolute_import, division, print_function)
+
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: emc_vnx_mv_promote
+
+short_description: Promote mirrorview Mirror or Consistency Group on EMC VNX
+
+version_added: "2.7"
+
+description:
+    - "This module manages promotion of a mirror view async connection."
+
+extends_documentation_fragment:
+  - emc.emc_vnx
+
+options:
+    name:
+        description:
+            - Name of the MirrorView Mirror or Consistency Group.
+        required: true
+    force:
+        description:
+            - Force promotion if sites are not connected (local promotion).
+        type: bool
+        required: false
+
+
+author:
+    - Luca 'remix_tj' Lorenzetto (@remixtj)
+'''
+
+EXAMPLES = '''
+- name: Promote MV/A
+  emc_vnx_mv_promote:
+    name: mg01
+    sp_user: sysadmin
+    sp_password: sysadmin
+    sp_address: sp1a.fqdn
+
+- name: Promote MV/A Locally
+  emc_vnx_mv_promote:
+    name: mg01
+    sp_user: sysadmin
+    sp_password: sysadmin
+    sp_address: sp1a.fqdn
+    force: True
+'''
+
+RETURN = '''
+promoted_mirrors:
+    description: list of the mirrors that has been promoted.
+    type: list
+    returned: success
+promoted_luns:
+    description: list of the luns that have been promoted.
+    type: list
+    returned: success
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.storage.emc.emc_vnx import emc_vnx_argument_spec
+
+
+try:
+    from storops import VNXSystem
+    from storops.exception import VNXCredentialError, VNXMirrorException
+    HAS_LIB = True
+except:
+    HAS_LIB = False
+
+
+def run_module():
+    module_args = dict(
+        name=dict(type='str', required=True),
+        force=dict(type='bool', required=False, default=False)
+    )
+
+    module_args.update(emc_vnx_argument_spec)
+
+    result = dict(
+        changed=False,
+        promoted_mirrors=[],
+        promoted_luns=[]
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
```

Comment:
It doesn't really support `check_mode` since when check mode is enabled `changed` is always `False`.